server.js

```diff
@@ -2,37 +2,134 @@ import express from 'express'
+  const restaurant = await Michelin.find();
+
+  if ( restaurant.length > 0  ) {
+    res.json(restaurant)
+  } else {
+    res.status(404).json({ message: `Error, no restaurants found.` })
+  }
+});
+
+// List all regions with /michelin/regions
+// Query region with /michelin/regions?region=sweden
+app.get('/michelin/regions/', async ( req, res ) => {
+  const { region } = req.query
+
+  let restaurantByRegion = await Michelin.find({ region })
+  let allUniqueRegions = await Michelin.find({}).distinct('region')
+
+  if ( restaurantByRegion.length > 0 ) {
+    res.json({ totalResults: restaurantByRegion.length, restaurantByRegion })
+  } else if ( region && restaurantByRegion.length === 0 ) {
+    res.status(404).json({ message: `Error, ${region} not found` })
+  }
+  res.json({ totalResults: allUniqueRegions.length, allUniqueRegions })
 })
 
-// Start the server
+// List all cuisines with /michelin/cuisines
+// Query cuisine with /michelin/cuisines?cuisine=creative
+app.get('/michelin/cuisines/', async ( req, res ) => {
+  const { cuisine } = req.query
+
+  let restaurantByCuisine = await Michelin.find({ cuisine })
+  let allUniqueCuisines = await Michelin.find({}).distinct('cuisine')
+
+  if ( restaurantByCuisine.length > 0 ) {
+    res.json({ totalResults: restaurantByCuisine.length , restaurantByCuisine })
+  } else if ( cuisine && restaurantByCuisine.length === 0 ) {
+    res.status(404).json({ message: `Error, ${cuisine} not found` })
+  }
+  res.json({ totalResults: allUniqueCuisines.length, allUniqueCuisines })
+})
+
+// List all citys with /michelin/citys
+// Query city with /michelin/citys?city=stockholm
+app.get('/michelin/citys/', async ( req, res ) => {
+  const { city } = req.query
+
+  let restaurantByCity = await Michelin.find({ city })
+  let allUniqueCitys = await Michelin.find({}).distinct('city')
+
+  if ( restaurantByCity.length > 0 ) {
+    res.json({ totalResults: restaurantByCity.length , restaurantByCity })
+  } else if ( city && restaurantByCity.length === 0 ) {
+    res.status(404).json({ message: `Error, no restaurants found in ${city}` })
+  }
+  res.json({ totalResults: allUniqueCitys.length, allUniqueCitys })
+})
+
+// Find restaurant by name
+// Use parameter with /michelin/pierre
+app.get('/michelin/:name', async ( req, res ) => {
+  const { name } = req.params
+  const restaurantByName = await Michelin.findOne({ name: name })
+  if ( restaurantByName ) {
+    res.json( restaurantByName )
+  } else {
+    res.status(404).json({ message: `Error, no restaurant by that name` })
+  }
+})
+
+// Find restaurants by year
+// Use parameter with /michelin/2019
+app.get('/michelin/year/:year', async ( req, res ) => {
+  const { year } = req.params
+  const restaurantByYear = await Michelin.find({ year: year })
+
+  if ( restaurantByYear.length > 0 ) {
+    res.json({ totalResults: restaurantByYear.length, restaurantByYear})
+  } else {
+    res.status(404).json({ message: `Error, no restaurants by that year` })
+  }
+})
```

Comment:
One advice to make it more RESTful is that you name your main endpoint descriptive in plural, for example `/restaurants`, and then you add queries for the parameters. The endpoints should be descriptive of what it returns and not how they are filtered (meaning, this endpoint doesn't return years, it filters on years). So you could have queries like this:
```
/restaurant?region=YOUR_COUNTRY
/restaurant?cuisine=YOUR_CUISINE
/restaurant?city=YOUR_CITY
/restaurant?name=SEARCH_NAME
/restaurant?year=SEARCH_YEAR

```